expression/chunk_executor.go

```diff
@@ -346,6 +346,22 @@ func VectorizedFilter(ctx sessionctx.Context, filters []Expression, iterator *ch
 // filters, `isNull` indicates whether the result of the filter is null.
 // Filters is executed vectorized.
 func VectorizedFilterConsiderNull(ctx sessionctx.Context, filters []Expression, iterator *chunk.Iterator4Chunk, selected []bool, isNull []bool) ([]bool, []bool, error) {
+	// canVectorized use to check whether all of the filters can vectorized evaluate by chunk
+	canVectorized := true
+	for _, filter := range filters {
+		if !filter.Vectorized() {
+			canVectorized = false
+			break
+		}
+	}
+	if canVectorized {
+		return vectorizedFilterConsiderNull2(ctx, filters, iterator, selected, nil)
+	}
+	return vectorizedFilterConsiderNull(ctx, filters, iterator, selected, nil)
+}
+
+// Filters can not be evaluated vectorized.
+func vectorizedFilterConsiderNull(ctx sessionctx.Context, filters []Expression, iterator *chunk.Iterator4Chunk, selected []bool, isNull []bool) ([]bool, []bool, error) {
```

Comment:
We can name these two functions more meaningfully. How about
`vectorizedFilterConsiderNull` -> `rowBasedFilter`
`vectorizedFilterConsiderNull2` -> `vectorizedFilter`

Reply:
Update. PTAL